### ISSUE TITLE
Note editor's spell checker doesn't seem to work for cyrillic languages

### ISSUE THREAD

**Issue by d1vanov** [d1vanov/libquentier#39]
Most probably it has something to do with Unicode i.e. with missing support for it in JavaScript. Not sure I'd be able to handle this but need to try at least.

**Comment by michaelkn** [comment]
Origin of the problem is found:
```
// spellChecker.js
    this.setRegex = function(input) {
        ...
        input = input.replace(/[^\w\\\s']+/g, "");
```
`\w` matches any alphanumeric character from the basic Latin alphabet, including the underscore. Hence `^\w` removes cyrillic symbols.

**Comment by d1vanov** [comment]
This one was fixed via PR #48.
